fog/block_provider/src/local.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2018-2023 The MobileCoin Foundation
+
+use crate::{
+    BlockDataWithTimestamp, BlockProvider, BlocksDataResponse, Error, TxOutInfoByPublicKeyResponse,
+};
+use mc_blockchain_types::{Block, BlockIndex};
+use mc_crypto_keys::CompressedRistrettoPublic;
+use mc_fog_api::ledger::{TxOutResult, TxOutResultCode};
+use mc_ledger_db::{Error as LedgerError, Ledger};
+use mc_transaction_core::tx::{TxOut, TxOutMembershipProof};
+use mc_watcher::watcher_db::WatcherDB;
+use mc_watcher_api::TimestampResultCode;
+use std::time::Duration;
+
+#[derive(Clone)]
+pub struct LocalBlockProvider<L: Ledger + Clone + Sync> {
+    ledger: L,
+    watcher: Option<WatcherDB>,
+}
+
+impl<L: Ledger + Clone + Sync> LocalBlockProvider<L> {
+    pub fn new(ledger: L, watcher: impl Into<Option<WatcherDB>>) -> Box<Self> {
+        Box::new(Self {
+            ledger,
+            watcher: watcher.into(),
+        })
+    }
+
+    fn get_tx_out_result(
+        &self,
+        tx_out_pubkey: &CompressedRistrettoPublic,
+    ) -> Result<TxOutResult, LedgerError> {
+        let mut result = TxOutResult::new();
+        result.set_tx_out_pubkey(tx_out_pubkey.into());
+
+        let tx_out_index = match self.ledger.get_tx_out_index_by_public_key(tx_out_pubkey) {
+            Ok(index) => index,
+            Err(LedgerError::NotFound) => {
+                result.result_code = TxOutResultCode::NotFound;
+                return Ok(result);
+            }
+            Err(err) => {
+                return Err(err);
+            }
+        };
+
+        result.result_code = TxOutResultCode::Found;
+        result.tx_out_global_index = tx_out_index;
+
+        let block_index = match self.ledger.get_block_index_by_tx_out_index(tx_out_index) {
+            Ok(index) => index,
+            Err(_err) => {
+                // TODO
```

Comment:
Currently there's no `Logger`. I am not sure it's worth adding it just for this log message...